docs/proposals/cluster-bootstrap-with-gossip.md

```diff
@@ -0,0 +1,153 @@
+# Draft Proposal: Cluster Bootstrap with Gossip
+
+Luke Marsden, Ilya Dmitrichenko @ Weaveworks
+
+4th August 2016
+
+*This is a sub-proposal of [Proposal: Dramatically Simplify Kubernetes Cluster Creation](https://github.com/kubernetes/kubernetes/pull/30360)*. Where "Phase I" and "Phase II" are referred to below, they refer to the phases defined in the parent proposal.
+
+## Motivation
+
+As part of the dramatically simplified cluster creation UX described in the above linked proposal, it is desirable to have a mechanism for discovery which enables the desired UX without depending on an external network service, such as a discovery service, so that we don’t have to operate one, and users don’t have to leave their firewall to provision a cluster.
```

Comment:
Maybe I missed the discussion around this, but I'm not convinced that it's valuable to avoid having a discovery meet-up point (by default) to simplify installation.